pkg/resource/deploy/source_query.go

```diff
@@ -332,6 +333,50 @@ func (rm *queryResmon) Cancel() error {
 	return <-rm.done
 }
 
+// getProviderReference fetches the provider reference for a resource, read, or invoke from the given package with the
+// given unparsed provider reference. If the unparsed provider reference is empty, this function returns a reference
+// to the default provider for the indicated package.
+func getProviderReference(defaultProviders *defaultProviders, req providers.ProviderRequest,
```

Comment:
So we're not going to support parameterized providers in `pulumi query`, which makes sense to me.  Should we at return an error response with a little bit more information if `RegisterProvider` is called? I think the answer is no, and whatever error message Grpc gives us in this case is good enough, but figured I ask the question anyway.

Reply:
> So we're not going to support parameterized providers

No intention for it correct.

> I think the answer is no, and whatever error message Grpc gives us in this case is good enough,

Yeh just "Not Implemented" is fine. The SDK will catch and handle that, needs to handle it for old engine as well anyway.